Articles/ShortArticles/MythOfDeveloperProductivity.md

```diff
@@ -0,0 +1,40 @@
+# Myth of Developer Productivity
+
+<!--deck text start-->
+The ability to directly effectively measure individual software developer productivity has eluded software developers for decades.  Why is this so hard and what are the alternatives? 
+<!--deck text end-->
+
+#### Contributed by [Roscoe A. Bartlett](https://github.com/bartlettroscoe), [Mark C. Miller](https://github.com/markcmiller86)
+
+#### Publication date: December, ??? 2022
+
+Here are the references I want to include: [myth-dev-prod], [cant-meas-prod], [kanban-and-scrum], [impl-lean-dev] (NOTE: We need a CC article for [impl-lean-dev].)
+
+Key points to make?
+
+* Why we care to measure productivity? => Software estimation, individual developer performance evaluation?
+
+* The number one way of improving productivity is to write less software (see [impl-lean-dev]).
+
+* The cost of software maintenance is often several times much higher than the cost to create the software in the first place.
+
+* The on-going cost of software maintenance is strongly correlated with the size of the code base (see: code complete???).
+
+* Defining productivity by directly measuring the amount of software created disincentivizes solutions that may reuse existing software or refactorings that may actaully result in less overall source code.
+
```

Comment:
Developers who remove code (or have a net negative contribution on GitHub) yet maintain or even enhance functionality should be celebrated. Are they?

We don't have good processes for removing obsoleted code. All of our processes are about *adding* and supporting code.

Reply:
I personally feel great when I am able to delete a bunch of code that I wrote years ago and can maintain needed functionality.  Every line of code deleted is one less line of code to test and maintain.

For example, making a code build faster and have its test suite run faster does not directly improve functionality but can greatly improve future development productivity.  (E.g. a single Trilinos PR build takes over 6 wall-clock hours to run, and that is once it has started running.  It can take days to get the PR tester to start running the builds.)